lib/eslint.js

```diff
@@ -894,7 +896,13 @@ module.exports = (function() {
             column: location.column + 1,   // switch to 1-base instead of 0-base
             nodeType: node.type,
             source: sourceCode.lines[location.line - 1] || ""
-        });
+        };
+
+        if (fix && typeof fix === "object") {
```

Comment:
Do we want to be more strict here about type checking like checking to make sure its not an Array?
Since this is exposed and will be used by code outside the project.